### PR TITLE
[#2723] Support multiple backends for Notifications API

### PR DESCRIPTION
**Pull request by pi-sigma**
Taiga: https://taiga.maykinmedia.nl/project/open-inwoner/task/2723

Related: https://taiga.maykinmedia.nl/project/open-inwoner/task/2700

Note: the `notificaties.test.openzaak` backend does not set an auth header when sending test notifications for newly created subscriptions, which can lead to false positive error logs when testing webhooks (cf. https://sentry.maykinmedia.nl/organizations/maykin-media/issues/363204/?project=284)

### CODE REVIEW

File: src/open_inwoner/conf/base.py [hunk 1]
```diff
@@ -237,12 +238,6 @@
     "djchoices",
     "django_celery_beat",
     "django_celery_monitor",
-    # Temporary fix: the notifications lib interferes with
-    # celery's task loading meachanism, which prevents certain
-    # tasks from showing up in the admin when OIP is run with
-    # Docker; this needs to be fixed this in the library eventually;
-    # for now we load it after all our apps.
-    "notifications_api_common",
 ]
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/open_inwoner/conf/base.py:241]: This is the implementation: https://github.com/open-formulieren/open-forms/blob/a4cfd74aa47a547c7a36055daf8611af128b7bba/src/openforms/upgrades/upgrade_paths.py#L78

Next week is Open Forms off-week - I could use that time to extract this code into a re-usable django-app/package that you could then use in OIP for these kind of migrations. Would that be acceptable?

File: src/open_inwoner/openzaak/api/views.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-from notifications_api_common.api.serializers import NotificatieSerializer
 from rest_framework import status
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from zgw_consumers.api_models.base import factory
 
+from notifications.api.serializers import NotificatieSerializer
 from open_inwoner.configurations.models import SiteConfiguration
 from open_inwoner.openzaak.api_models import Notification
 from open_inwoner.openzaak.auth import get_valid_subscription_from_request
```
[the remark this answers is not in the document]
Comment (pi-sigma) [reply src/open_inwoner/openzaak/api/views.py:11]: Checked: the only lingering references to `notifications_api_common` are in `INSTALLED_APPS`, the data migration and the associated test.

I am not sure we need special logic in the `NotificationsWebhookBaseView` to handle different backends (in the way we need special logic to handle differen ZGW backends). We need to be able to register webhooks with different backends, but this is now possible. When a request comes in at the  `NotificationsWebhookBaseView`, we determine the subscription from the request, create a notification from the data and pass it on to the notification handler.

File: src/notifications/admin.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        try:
+            sub.register()
+        except RequestException as exc:
+            messages.error(
+                request,
+                _(
+                    "Something went wrong while registering subscription "
+                    "for {callback}: {exception}"
+                ).format(callback=sub.callback_url, exception=exc),
```
[the remark this answers is not in the document]
Comment (pi-sigma) [reply src/notifications/admin.py:27]: Discussed: we use old-school (`%`-style) interpolation for logging (to support log aggregation), and `.format`-style interpolation for translations in messages

File: src/notifications/models.py [hunk 4]
```diff
@@ -131,6 +131,7 @@ def register(self) -> None:
         }
 
         response = client.post("abonnement", json=data)
+        response.raise_for_status()
```
[the remark this answers is not in the document]
Comment (pi-sigma) [reply src/notifications/models.py:134]: That's the default, the error message is displayed in the admin

**Pull request merged**